GitHub workflow for publishing documentation is failing

**Describe the bug**

```
+ cd docs
+ make html
Running Sphinx v7.3.7

Extension error:
Could not import extension IPython.sphinxext.ipython_directive (exception: cannot import name 'Self' from 'typing_extensions' (/home/runner/work/datafusion-python/datafusion-python/venv/lib/python3.10/site-packages/typing_extensions.py))
make: *** [Makefile:38: html] Error 2
```

**To Reproduce**

**Expected behavior**

**Additional context**

--- a/.github/workflows/docs.yaml
+++ b/.github/workflows/docs.yaml
@@ -48,7 +48,7 @@ jobs:
           set -x
           python3 -m venv venv
           source venv/bin/activate
-          pip install -r requirements-310.txt
+          pip install -r requirements-311.txt
           pip install -r docs/requirements.txt
       - name: Build Datafusion
         run: |

